server/channels/app/export.go

```diff
@@ -878,14 +882,38 @@ func (a *App) exportAllDirectChannels(ctx request.CTX, job *model.Job, writer io
 		for _, channel := range channels {
 			afterId = channel.Id
 
+			// Skip deleted.
+			if channel.DeleteAt != 0 {
+				continue
+			}
+
 			// Skip if there are no active members in the channel
 			if len(channel.Members) == 0 {
 				continue
 			}
 
-			// Skip deleted.
-			if channel.DeleteAt != 0 {
-				continue
+			// Skip if the channel member structure is not intact
+			switch channel.Type {
+			case model.ChannelTypeGroup:
+				groupMembers := make([]string, len(channel.Members))
+				for i, m := range channel.Members {
+					groupMembers[i] = m.UserId
+				}
+				if channel.Name != model.GetGroupNameFromUserIds(groupMembers) {
+					// this is the case of a group channel when other user is permanently deleted
+					// we skip this channel and inform by logging it.
+					job.Data["skipped_direct_channels"] = job.Data["skipped_direct_channels"] + "," + channel.Id
+					ctx.Logger().Info("Skipping unintegrated group channel", mlog.String("channel_id", channel.Id))
```

Comment:
"unintegrated" sounds like an odd choice of word. I'd rather say "Skipping GM channel with partially deleted members", and also log it an WARN level while we're at it, because it's a change in behavior.